hack/kserve_migration/kserve_migration.sh

```diff
@@ -0,0 +1,236 @@
+                    "${prefix}-controller-manager-service" \
+                    "${prefix}-models-web-app" \
+                    "${prefix}-webhook-server-service"; do
+        if [ ! -z "${svc_names##*$svc_name*}" ]; then
+            log ERROR "${prefix} controller services are not installed completely."
+            exit 1;
+        fi
+    done
+    po_names=$(kubectl get po -n $namespace -o jsonpath='{.items[*].metadata.name}')
+    for po_name in "${prefix}-controller-manager" "${prefix}-models-web-app"; do
+        if [ ! -z "${po_names##*$po_name*}" ]; then
+            log ERROR "${prefix} controller services are not installed completely."
+            exit 1;
+        fi
+    done
+}
+
+# Checks user preference on cleaning kfserving controller
+if [ "${REMOVE_KFSERVING}" == "false" ]; then
+    CLEAN_KFSERVING="false"
+fi
+
+# Checks whether the kfserving is running or not
+log INFO "checking whether kfserving is running or not"
+isControllerRunning kfserving-system
+
+# Checks whether the kserve is running or not
+log INFO "checking whether kserve is running or not"
+isControllerRunning kserve
+
+# # Deploy kserve
+# log INFO "deploying kserve"
+# cd ..
+# KSERVE_CONFIG=kserve.yaml
+# for i in 1 2 3 4 5 ; do kubectl apply -f install/${KSERVE_VERSION}/${KSERVE_CONFIG} && break || sleep 15; done
+# kubectl wait --for=condition=ready --timeout=120s po --all -n kserve
+# isControllerRunning kserve
+# cd hack
+# log INFO "kserve deployment completed"
+
+# Get inference services config
+log INFO "getting inference services config"
+inference_services=$(kubectl get inferenceservice.serving.kubeflow.org -A -o jsonpath='{.items[*].metadata.namespace},{.items[*].metadata.name}')
+declare -a isvc_names
+declare -a isvc_ns
+if [ ! -z "$inference_services" ]; then
+    mkdir -p ${ISVC_CONFIG_DIR}
+    IFS=','; isvc_split=($inference_services); unset IFS;
+    isvc_ns=(${isvc_split[0]})
+    isvc_names=(${isvc_split[1]})
+fi
+isvc_count=${#isvc_names[@]}
+for (( i=0; i<${isvc_count}; i++ ));
+do
+    kubectl get inferenceservice.serving.kubeflow.org ${isvc_names[$i]} -n ${isvc_ns[$i]} -o yaml > "${ISVC_CONFIG_DIR}/${isvc_names[$i]}.yaml"
+done
+
+# Get knative services names
+log INFO "getting knative services"
+knative_services=$(kubectl get ksvc -A -o jsonpath='{.items[*].metadata.namespace},{.items[*].metadata.name}')
+declare -a ksvc_names;
+declare -a ksvc_ns;
+if [ ! -z "$knative_services" ]; then
+    mkdir -p ${KSVC_CONFIG_DIR}
+    IFS=','; ksvc_split=(${knative_services}); unset IFS;
+    ksvc_ns=(${ksvc_split[0]})
+    ksvc_names=(${ksvc_split[1]})
+fi
+ksvc_count=${#ksvc_names[@]}
+
+(
+    # Stop kfserving controller
+    log INFO "stopping kfserving controller"
+    kubectl scale --replicas=0 statefulset.apps kfserving-controller-manager -n kfserving-system
+    sleep 30
+
+    trap 'kubectl scale --replicas=1 statefulset.apps kfserving-controller-manager -n kfserving-system' ERR
```

Comment:
what does this do?

Reply:
@yuzisun If we face any issues in deploying new inference service, It restores kfserving controller.

Reply:
I think we should only have one controller running, if we scale back `kfserving-controller` then we probably should scale down `kserve-controller`